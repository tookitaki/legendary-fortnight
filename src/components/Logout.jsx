import PropTypes from 'prop-types';
import { Button } from 'antd';
import { connect } from 'react-redux';
import { logoutUser } from '../actions/login';

export function Logout({ loading, dispatchLogoutUser }) {
  const handleLogout = (e) => {
    e.preventDefault();
    dispatchLogoutUser();
  };

  return (
    <Button
      id="login"
      data-testid="login"
      disabled={loading}
      type="primary"
      htmlType="submit"
      loading={loading}
      block
      onClick={handleLogout}>
      Logout
    </Button>
  );
}

function mapDispatchToProps(dispatch) {
  return {
    dispatchLogoutUser: () => {
      dispatch(logoutUser());
    }
  };
}

Logout.propTypes = {
  dispatchLogoutUser: PropTypes.func,
  loading: PropTypes.bool
};

export default connect(null, mapDispatchToProps)(Logout);
