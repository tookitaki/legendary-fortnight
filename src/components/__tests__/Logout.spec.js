import { Logout } from '../Logout';
import { fireEvent, render } from '@testing-library/react';

describe('<Logout />', () => {
  let props, component;
  beforeEach(() => {
    props = {
      dispatchLogoutUser: jest.fn()
    };
    component = render(<Logout {...props} />);
  });

  it('should render logout snapshot', () => {
    expect(component.container).toMatchSnapshot();
  });

  it('should call login on click of LOGIN button', () => {
    const { getByTestId } = component;

    const logout = getByTestId('login');
    fireEvent.click(logout, { preventDefault: jest.fn() });
    expect(props.dispatchLogoutUser).toHaveBeenCalledTimes(1);
  });
});
