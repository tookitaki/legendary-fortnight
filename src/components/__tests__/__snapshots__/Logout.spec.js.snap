// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout /> should render logout snapshot 1`] = `
<div>
  <button
    class="ant-btn ant-btn-primary ant-btn-block"
    data-testid="login"
    id="login"
    type="submit"
  >
    <span>
      Logout
    </span>
  </button>
</div>
`;
